src/Cpp/1-getting-started/14-HelloCOM/HelloCOM.cpp

```diff
@@ -0,0 +1,330 @@
+#include "HelloCOM.hpp"
+
+#include <GLFW/glfw3.h>
+#define GLFW_EXPOSE_NATIVE_WIN32
+#include <GLFW/glfw3native.h>
+
+#include <d3dcompiler.h>
+
+#include <iostream>
+
+#pragma comment(lib, "d3d11.lib")
+#pragma comment(lib, "dxgi.lib")
+#pragma comment(lib, "d3dcompiler.lib")
+#pragma comment(lib, "winmm.lib")
+#pragma comment(lib, "dxguid.lib")
+
+struct VertexFormat
+{
+    float vertex[3];
+    float color[3];
+};
+
+HelloCOMApplication::HelloCOMApplication(const std::string_view title)
+    : Application(title)
+{
+}
+
+HelloCOMApplication::~HelloCOMApplication()
+{
+    _deviceContext->Flush();
+    _pixelShader.Reset();
+    _vertexShader.Reset();
+    _vertexLayout.Reset();
+    _triangleIndices.Reset();
+    _triangleVerts.Reset();
+    _renderTarget.Reset();
+    _swapChain.Reset();
+    _dxgiFactory.Reset();
+#if !defined(NDEBUG)
+    _debug->ReportLiveDeviceObjects(D3D11_RLDO_SUMMARY);
+    _debug.Reset();
+#endif
+    _deviceContext.Reset();
+    _device.Reset();
+    Application::Cleanup();
+}
+
+bool HelloCOMApplication::Initialize()
+{
+    // This section initializes GLFW and creates a Window
+    if (!Application::Initialize())
+    {
+        return false;
+    }
+    glfwSetWindowUserPointer(GetWindow(), this);
```

Comment:
Hmm yeah, I fink, I would like this glfw stuff contained in `Application`.

Could you introduce a virtual method in `Application` like `WindowResized` set it up as the callback there, and just overwrite `WindowResized` here in HelloXXX. That would keep dependencies separate